File: android/src/main/java/com/getcapacitor/community/classes/options/SendPayloadOptions.java
package com.getcapacitor.community.classes.options;

import android.util.Base64;
import androidx.annotation.Nullable;
import com.getcapacitor.JSArray;
import com.getcapacitor.PluginCall;
import java.util.ArrayList;
import java.util.List;
import org.json.JSONException;

public class SendPayloadOptions {

    @Nullable
    private List<String> endpointIDs;

    @Nullable
    private byte[] payload;

    public SendPayloadOptions(PluginCall call) throws JSONException {
        List<String> endpointIDs = new ArrayList<>();

        @Nullable
        String endpointID = call.getString("endpointID");
        if (endpointID != null && !endpointID.isEmpty()) {
            endpointIDs.add(endpointID);
        }

        @Nullable
        JSArray endpointArray = call.getArray("endpointIDs");
        if (endpointArray != null) {
            for (var item : endpointArray.toList()) {
                if (endpointID != null && !endpointID.isEmpty()) {
                    endpointIDs.add((String) item);
                }
            }
        }

        this.setEndpointIDs(endpointIDs);

        @Nullable
        String payload = call.getString("payload");
        this.setPayload(payload);
    }

    public void setEndpointIDs(@Nullable List<String> endpointIDs) {
        this.endpointIDs = endpointIDs;
    }

    public void setPayload(@Nullable byte[] payload) {
        this.payload = payload;
    }

    public void setPayload(@Nullable String payload) {
        this.payload = (payload == null) ? null : Base64.decode(payload, Base64.NO_WRAP);
    }

    @Nullable
    public List<String> getEndpointIDs() {
        return endpointIDs;
    }

    @Nullable
    public byte[] getPayload() {
        return payload;
    }
}

File: ios/Sources/NearbyPlugin/Classes/Options/SendPayloadOptions.swift
import Foundation
import Capacitor

import NearbyConnections

@objc public class SendPayloadOptions: NSObject {
    private var endpointIDs: [EndpointID]?

    private var payload: Data?

    init(_ call: CAPPluginCall) {
        var endpointIDs: [EndpointID] = []

        if let endpointID = call.getString("endpointID") {
            endpointIDs.append(endpointID)
        }

        if let endpointArray = call.getArray("endpointIDs") {
            endpointIDs.append(contentsOf: endpointArray.capacitor.replacingNullValues() as! [EndpointID])
        }

        if endpointIDs.isEmpty {} else {
            self.endpointIDs = endpointIDs
        }

        if let payload = call.getString("payload") {
            self.payload = payload.data(using: .utf8)
        }
    }

    func getEndpointIDs() -> [EndpointID]? {
        return endpointIDs
    }

    func getPayload() -> Data? {
        return payload
    }
}
